### DISCUSSION TITLE
Connection with ssh-agent

### DISCUSSION THREAD

**Discussion by CoderMcDaniel** [smallstep/certificates#1941] [Q&A] [answered]
I am using the 0.26.2 step-ca container. I have successfully been able to bootstrap a client (step-cli container 0.26.2).  However on a large portion of the step-cli commands I get an error stating that I cannot connect to the SSH agent.  I confirmed that the host (RHEL8) I am running the CLI container on has an SSH agent.  How should I configure the container to interact with the SSH agent?

Thanks for any help!

**Comment by hslatman** [comment, marked as answer]
Hey @CoderMcDaniel,

If you run the SSH agent on your host, and want `step` in a Docker container to interface with it, you'll have to share the SSH agent socket with the Docker container. This blog explains how you can do so: https://www.jamesridgway.co.uk/sharing-an-ssh-agent-between-a-host-machine-and-a-docker-container/.

Generally users run `step` natively on their machine, so that `step` can interact with the SSH agent directly. That way you don't need to share the SSH agent socket with the Docker container.

**Reply by CoderMcDaniel** [reply]
@hslatman Thanks for the help.  That is somewhat what I figured.  

I did see this article before, but did not see SSH_AUTH_SOCK environment variable set.  Do I need to evaluate the return of 'ssh-agent'? "eval ssh-agent"?
